### ISSUE TITLE
Spurious unused type parameter warnings

### ISSUE THREAD

**Issue by WeeknightMVP** [GaloisInc/saw-script#242]
Thank you for pointing out my errors in SAW scripts for issues #240 and (Crucible) [#33](https://github.com/GaloisInc/crucible/issues/33).

After generalizing `clear_void_spec` to take a size parameter, SAW reports that the parameter `tp_size` is unused, but it is used and the proof succeeds:

```Haskell
let clear_void_spec (tp_size : Int) : CrucibleSetup() = do {
    let voidArrayType = (llvm_array tp_size (llvm_int 8));
    arr <- (crucible_fresh_var "arr" voidArrayType);
    let v_arr = (crucible_term arr);
    p_arr <- (crucible_alloc voidArrayType);
    let v_size = (crucible_term {{ `tp_size:[32] }}); 
    crucible_points_to p_arr v_arr;
    crucible_execute_func [p_arr, v_size];
    crucible_points_to p_arr (crucible_term {{ zero:[tp_size][8] }});
};

let main : TopLevel () = do {
    voidTest <- llvm_load_module "voidTest.bc";
    clear_void_12_result <- crucible_llvm_verify voidTest "clear_void" [] false (clear_void_spec 12) z3;
    print "Done!";
};
```

> \> saw voidTest.saw
> Loading file "...voidTest.saw"
> [warning] at :1:1--1:1 Unused name: tp_size
> [warning] at :1:1--1:1 Unused name: tp_size
> Proof succeeded! @clear_void
> Done!

This is fairly recent; the 31 Oct 2017 nightly succeeds without warnings (but fails with a bad pointer load when trying to use it to prove `clear_uints`, whereas the 5 Dec nightly succeeds and reports spurious unused name warnings).

**Comment by brianhuffman** [comment]
Here's a smaller example that gives the same warning:
```
sawscript> let f (n:Int) = {{ 0 : [n] }}
sawscript> f 5
[warning] at :1:1--1:1 Unused name: n
Prelude.bvNat 5 0
```
This warning message was added to the Cryptol typechecker by @yav in revision GaloisInc/cryptol@5ccb0bb4. The warning is supposed to be helpful if you're writing a parameterized module and have a type parameter that is never used.

In saw-script, when we call function `f` the parameter `(n:Int)` gets added to the Cryptol environment as a type synonym. E.g. if the argument is `5`, then we add the declaration `type n = 5`. Then we use this environment to parse the string `0 : [n]`. I don't know yet why we get the warning, as we actually *are* using `n` in this case.

**Comment by brianhuffman** [comment]
Fixed in GaloisInc/cryptol@60e03d731346207670b6e2a2519174a1e3976193 and 6dde9970ac3823c1c108cc0d172591c0424edc6e.
